src/libData/AccountData/AccountStore.cpp

```diff
@@ -608,5 +608,31 @@ bool AccountStore::MigrateContractStates2(
     return false;
   }
 
+  return true;
+}
+
+bool AccountStore::DumpCode() {
+  LOG_MARKER();
+
+  for (const auto& i : m_state) {
+    Address address(i.first);
+
+    Account account;
+    if (!account.DeserializeBase(bytes(i.second.begin(), i.second.end()), 0)) {
+      LOG_GENERAL(WARNING, "Account::DeserializeBase failed");
+      return false;
+    }
+
+    if (account.isContract()) {
+      account.SetAddress(address);
+      std::ofstream os("contracts/" + account.GetAddress().hex() +
+                       CONTRACT_FILE_EXTENSION);
```

Comment:
We'll very soon be able to release the external libraries feature, so this should be something like

```
(is_library ? LIBRARY_CODE_EXTENSION : CONTRACT_FILE_EXTENSION)
```
I can't remember how to determine `is_library` from an Account, but I've seen logic that determines it somewhere else, so it must be possible.

Reply:
oh okay, it's possible but may need some extra effort to do so. but seems we haven't library code yet, maybe we can live with this first?

Reply:
Thanks.